script.js

```diff
@@ -1,12 +1,235 @@
+    </section>`;
+
+      let seasonNumber = addZero(episodes[i].season);
+      let episodeNumber = addZero(episodes[i].number);
+      optionElement.innerHTML = `S${seasonNumber}E${episodeNumber} - ${episodes[i].name}`;
+      selectEpisodeElement.appendChild(optionElement);
+    }
+    selectEpisodeElement.addEventListener("change", () => {
+      if (selectEpisodeElement.selectedIndex === 0) {
+        selectedShow();
+      } else {
+        reset();
+        mainElement.innerHTML = selectEpisodeElement.value;
+        rootElem.appendChild(mainElement);
+      }
+    });
+  }
+}
+
+//select show
+let selectShow = function () {
+  mainElement.innerHTML = "";
+  const showList = getAllShows();
+  showList.sort((a, b) => (a.name > b.name ? 1 : -1));
+  var showArr = [];
+  var idArr = [];
+  for (let i = 0; i < showList.length; i++) {
+    showArr.push(showList[i].name);
+    idArr.push(showList[i].id);
+  }
+  for (let i = 0; i < showList.length; i++) {
+    let optionElement = document.createElement("option");
+    optionElement.value = `${idArr[i]}`;
+    optionElement.innerHTML = `${showArr[i]}`;
+    selectShowElement.appendChild(optionElement);
+  }
+};
+
+//adding event listener to choose selected show from select show option
+selectShowElement.addEventListener("change", selectedShow);
+
+function selectedShow() {
+  reset();
+  if (selectShowElement.selectedIndex === 0) {
+    mainElement.innerHTML = "";
+    searchBox("Shows");
+    displayShow();
+  } else {
+    mainElement.innerHTML = "";
+    let showId = selectShowElement.value;
+    selectEpisodeElement.innerHTML = "";
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("something wrong");
+        }
+        return response.json();
+      })
+      .then((data) => {
+        const allEpisodes = data;
+        dropDown(allEpisodes);
+        makePageForEpisodes(allEpisodes);
+        searchBox("Episodes");
+      })
+      .catch((error) => console.log(error));
+  }
+}
+
+// display All show on main page
+function displayShow() {
+  let showObj = getAllShows();
+  selectEpisodeElement.style.display = "none";
+  selectEpisodesLabel.style.display = "none";
+  for (let i = 0; i < showObj.length; i++) {
+    truncate;
+    showObj.sort((a, b) => (a.name > b.name ? 1 : -1));
+    let showDiv = document.createElement("div");
+    showDiv.className = "showSection";
+    if (showObj[i].image !== null) {
+      showDiv.innerHTML = `<div class=" show d-flex flex-column align-items-center mt-4">
+         <div class=" d-flex flex-column align-items-center justify-content-center">
+            <h4 class="pb-2" role="button">${showObj[i].name}</h4>
+            <img class="w-100 bd-highlight img-thumbnail" src =${showObj[i].image.medium}>
+        </div>
+        <div class="d-flex justify-content-between">
+            <div class= "summary p-4">Summary: ${showObj[i].summary}</div>
```

Comment:
You've accessed `showObj[i]` 8 times here. In a situation where the array is very large, it will be quite expensive to look up the same array item 8 times and so will slow your application down. It's faster to assign it to an array, i.e. `const show = showObj[i];` then use `show` as much as you need to.